lms/djangoapps/courseware/tests/test_credit_requirements.py

```diff
@@ -133,6 +136,34 @@ def test_credit_requirements_not_eligible(self):
         )
         self.assertContains(response, "Verification Failed")
 
+    @ddt.data(
+        (CourseMode.VERIFIED, True),
+        (CourseMode.CREDIT_MODE, True),
+        (CourseMode.HONOR, False),
+        (CourseMode.AUDIT, False),
+        (CourseMode.PROFESSIONAL, False),
+        (CourseMode.NO_ID_PROFESSIONAL_MODE, False)
+    )
+    @ddt.unpack
+    def test_credit_requirements_on_progress_page(self, enrollment_mode, display_requirement):
+        # Test the progress table is only displayed to the
+        # verified and credit students.
+        self.enrollment.mode = enrollment_mode
+        self.enrollment.save()  # pylint: disable=no-member
+
+        # Check the progress page display
+        response = self._get_progress_page()
+        if display_requirement:
```

Comment:
This can be simplified.

```python
# Verify the appropriate messaging is shown if the user is in a credit-eligible mode, and hidden otherwise.
classes = ('credit-eligibility', 'eligibility-heading',)
method = self.assertContains if display_requirement else self.assertNotContains

for _class in classes:
  method(response, _class)
```